src/main/java/com/example/travelservice/services/TravelerServiceImpl.java

```diff
@@ -41,5 +41,13 @@ public Traveler createTraveler(Traveler traveler) {
         return travelerRepository.save(traveler);
     }
 
-
+    @Override
+    public Traveler updateTraveler(Long id, Traveler traveler) {
+        traveler.setId(id);
+        if(travelerRepository.findById(id).isPresent()){
+            return travelerRepository.save(traveler);
+        } else{
+            return createTraveler(traveler);
```

Comment:
throw an exception instead of creating a new one

Reply:
logic changed